### DISCUSSION TITLE
Update Product if Product already exist

### DISCUSSION THREAD

**Discussion by JanikHalder** [swellstores/community#50] [Help] [answered]
Hi there,

I want to import products into swell from my PIM on a regular base.
Therefore I want to create products (post request) and if it fails because the PIM-ID (custom field in swell) is not unique, I want to update this product (put request). 
Is there a way to do this? Because as the error looks like that:
{
  errors: {
    sku: { message: 'Already exists', code: 'UNIQUE', fields: [Object] },
    slug: { message: 'Already exists', code: 'UNIQUE', fields: [Object] }
  }
}
there is no ID returned from the post request, which I can than use to update the product.

**Comment by JanikHalder** [comment, marked as answer]
I found an easy solution for this problem.

You can check if a product exist by using:

const product = await swell.get('/products', {
        where: {
            uniqueIdentifier: value,
        },
        limit: 25,
        page: 1,
    })

this returns

{
  count: numberOfItems,
  results: [items],
 page: 1,
}

Now you can check if numberOfItems === 0

If it is zero you have to create a new Product,
if it is 1 you get the id of the product by using product.results[0].id and now you can use this id to update the product.

Hope this helps.

Best, Janik

**Reply by ericingram** [reply]
There's another approach also, using the `slug` field to do a PUT request in a way that works to either create or update the record if it already exists. Here's an example:

```
await swell.put('/products/{slug}', {
  slug: 'new-or-existing',
  ...
})
```
